src/auth.ts

```diff
@@ -548,6 +548,85 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
   }
 }
 
+export abstract class NativeAuthType {
+  protected config: IConfig;
+  protected client: IClient;
+
+  constructor(deps: AuthTypeDependencies) {
+    this.config = deps.config;
+    this.client = deps.client;
+  }
+}
+
+export class GoogleNativeAuth extends NativeAuthType {
+  public authenticate(data?: any, options?: any): Promise<any> {
+    let deferred = new DeferredPromise<any, Error>();
+    const clientID = this.config.settings.nativeAuth['webClientId'];
+
+    if (!clientID) {
+      deferred.reject(new Error('Missing google web client id. Please visit http://docs.ionic.io/services/users/google-auth.html#native'));
+    }
+
+    GooglePlus.login({'webClientId': clientID, 'offline': true}).then((success) => {
```

Comment:
If you give a config option for facebook fields, you should give a config option for google scopes. 

https://github.com/EddyVerbruggen/cordova-plugin-googleplus#usage